### ISSUE TITLE
GitHub pages deploy before benchmarks are complete

### ISSUE THREAD

**Issue by moltar** [moltar/typescript-runtime-type-benchmarks#821]
I don't know if there is a way to specify the dependency on the GitHub actions to be completed.

But currently GH pages deploy in parallel, and before any benchmarks are done.

Ideas?

**Comment by moltar** [comment]
Maybe we can just trigger a rebuild as a hack?

https://gist.github.com/greghaskins/62e30796021e02869240

**Comment by hoeck** [comment]
But after benchmarks are completed, they commit the result and thus trigger a new pages deploy already?

So the only thing gained from that would be fewer gh-pages deploys. GH-docs say that there is a 10 pages builds per hour limit. Or am I missing something?

Are you asking because the benchmark results do not show up in the map? Sorry, that is my fault :sweat_smile:. The results json files contain a literal NUMBER-x as their `nodeVersion`. In app.tsx I am expecting plain nodejs vMajor-Minor-Path version strings.
This breaks the simple regex based filtering logic, not showing any results at all.

Two possible fixes:
1. fix the regex to deal with `"17.x"` and `"17.12.3"` version strings.
2. do not overwrite `NODE_VERSION` when running the benchmark in the release action (I would prefer this).

Should I open another PR or just commit into master?

**Comment by moltar** [comment]
> But after benchmarks are completed, they commit the result and thus trigger a new pages deploy already?

Oh yes, they do!

I assumed they wouldn't. Because GH has a policy of not running GH actions triggered from the result of a GH Action action (i.e. commit).

**Comment by moltar** [comment]
> Should I open another PR or just commit into master?

master is fine!

**Comment by hoeck** [comment]
> > But after benchmarks are completed, they commit the result and thus trigger a new pages deploy already?
> 
> Oh yes, they do!
> 
> I assumed they wouldn't. Because GH has a policy of not running GH actions triggered from the result of a GH Action action (i.e. commit).

Ok, didn't know that just looked at the action logs.

If this ever becomes a problem, it looks like disabling the default pages-build-and-deploy action and adding an explicit "deploy pages" step to the release like this one: https://github.com/marketplace/actions/github-pages-action would be the way to go.

> master is fine!
Ok, will fix the actions to use the full node version.
